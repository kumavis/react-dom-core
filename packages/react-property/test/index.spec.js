const DOMProperty = require('react-dom/lib/DOMProperty');
const HTMLDOMPropertyConfig = require('react-dom/lib/HTMLDOMPropertyConfig');
const main = require('..');

describe('HTMLDOMPropertyConfig', () => {
  DOMProperty.injection.injectDOMPropertyConfig(HTMLDOMPropertyConfig);

  it.each(Object.keys(DOMProperty.properties))(
    'matches property `%s` from react-dom',
    propName => {
      const property = DOMProperty.properties[propName];
      delete property.attributeNamespace;
      delete property.mutationMethod;
      const attributeName =
        HTMLDOMPropertyConfig.DOMAttributeNames[propName] ||
        propName.toLowerCase();
      expect(main.HTMLDOMPropertyConfig[attributeName]).toEqual(property);
    }
  );
});

describe('SVGDOMPropertyConfig', () => {
  it('matches snapshot', () => {
    expect(main.SVGDOMPropertyConfig).toMatchSnapshot();
  });
});

describe('isCustomAttribute', () => {
  it.each`
    argument                | expected
    ${''}                   | ${false}
    ${'dataaria'}           | ${false}
    ${'aria'}               | ${false}
    ${'aria-'}              | ${true}
    ${'aria-live'}          | ${true}
    ${'aria-live="polite"'} | ${false}
    ${'data'}               | ${false}
    ${'data-'}              | ${true}
    ${'data-foo'}           | ${true}
    ${'data-foo="bar"'}     | ${false}
  `(
    'returns $expected when argument is "$argument"',
    ({ argument, expected }) => {
      expect(main.isCustomAttribute(argument)).toBe(expected);
    }
  );
});
