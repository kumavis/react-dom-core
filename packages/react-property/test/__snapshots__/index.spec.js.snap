// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGDOMPropertyConfig matches snapshot 1`] = `
Object {
  "accent-height": Object {
    "propertyName": "accentHeight",
  },
  "accentheight": Object {
    "propertyName": "accentHeight",
  },
  "accumulate": Object {
    "propertyName": "accumulate",
  },
  "additive": Object {
    "propertyName": "additive",
  },
  "alignment-baseline": Object {
    "propertyName": "alignmentBaseline",
  },
  "alignmentbaseline": Object {
    "propertyName": "alignmentBaseline",
  },
  "allowReorder": Object {
    "propertyName": "allowReorder",
  },
  "allowreorder": Object {
    "propertyName": "allowReorder",
  },
  "alphabetic": Object {
    "propertyName": "alphabetic",
  },
  "amplitude": Object {
    "propertyName": "amplitude",
  },
  "arabic-form": Object {
    "propertyName": "arabicForm",
  },
  "arabicform": Object {
    "propertyName": "arabicForm",
  },
  "ascent": Object {
    "propertyName": "ascent",
  },
  "attributeName": Object {
    "propertyName": "attributeName",
  },
  "attributeType": Object {
    "propertyName": "attributeType",
  },
  "attributename": Object {
    "propertyName": "attributeName",
  },
  "attributetype": Object {
    "propertyName": "attributeType",
  },
  "autoReverse": Object {
    "propertyName": "autoReverse",
  },
  "autoreverse": Object {
    "propertyName": "autoReverse",
  },
  "azimuth": Object {
    "propertyName": "azimuth",
  },
  "baseFrequency": Object {
    "propertyName": "baseFrequency",
  },
  "baseProfile": Object {
    "propertyName": "baseProfile",
  },
  "basefrequency": Object {
    "propertyName": "baseFrequency",
  },
  "baseline-shift": Object {
    "propertyName": "baselineShift",
  },
  "baselineshift": Object {
    "propertyName": "baselineShift",
  },
  "baseprofile": Object {
    "propertyName": "baseProfile",
  },
  "bbox": Object {
    "propertyName": "bbox",
  },
  "begin": Object {
    "propertyName": "begin",
  },
  "bias": Object {
    "propertyName": "bias",
  },
  "by": Object {
    "propertyName": "by",
  },
  "calcMode": Object {
    "propertyName": "calcMode",
  },
  "calcmode": Object {
    "propertyName": "calcMode",
  },
  "cap-height": Object {
    "propertyName": "capHeight",
  },
  "capheight": Object {
    "propertyName": "capHeight",
  },
  "clip": Object {
    "propertyName": "clip",
  },
  "clip-path": Object {
    "propertyName": "clipPath",
  },
  "clip-rule": Object {
    "propertyName": "clipRule",
  },
  "clipPathUnits": Object {
    "propertyName": "clipPathUnits",
  },
  "clippath": Object {
    "propertyName": "clipPath",
  },
  "clippathunits": Object {
    "propertyName": "clipPathUnits",
  },
  "cliprule": Object {
    "propertyName": "clipRule",
  },
  "color-interpolation": Object {
    "propertyName": "colorInterpolation",
  },
  "color-interpolation-filters": Object {
    "propertyName": "colorInterpolationFilters",
  },
  "color-profile": Object {
    "propertyName": "colorProfile",
  },
  "color-rendering": Object {
    "propertyName": "colorRendering",
  },
  "colorinterpolation": Object {
    "propertyName": "colorInterpolation",
  },
  "colorinterpolationfilters": Object {
    "propertyName": "colorInterpolationFilters",
  },
  "colorprofile": Object {
    "propertyName": "colorProfile",
  },
  "colorrendering": Object {
    "propertyName": "colorRendering",
  },
  "contentScriptType": Object {
    "propertyName": "contentScriptType",
  },
  "contentStyleType": Object {
    "propertyName": "contentStyleType",
  },
  "contentscripttype": Object {
    "propertyName": "contentScriptType",
  },
  "contentstyletype": Object {
    "propertyName": "contentStyleType",
  },
  "cursor": Object {
    "propertyName": "cursor",
  },
  "cx": Object {
    "propertyName": "cx",
  },
  "cy": Object {
    "propertyName": "cy",
  },
  "d": Object {
    "propertyName": "d",
  },
  "decelerate": Object {
    "propertyName": "decelerate",
  },
  "descent": Object {
    "propertyName": "descent",
  },
  "diffuseConstant": Object {
    "propertyName": "diffuseConstant",
  },
  "diffuseconstant": Object {
    "propertyName": "diffuseConstant",
  },
  "direction": Object {
    "propertyName": "direction",
  },
  "display": Object {
    "propertyName": "display",
  },
  "divisor": Object {
    "propertyName": "divisor",
  },
  "dominant-baseline": Object {
    "propertyName": "dominantBaseline",
  },
  "dominantbaseline": Object {
    "propertyName": "dominantBaseline",
  },
  "dur": Object {
    "propertyName": "dur",
  },
  "dx": Object {
    "propertyName": "dx",
  },
  "dy": Object {
    "propertyName": "dy",
  },
  "edgeMode": Object {
    "propertyName": "edgeMode",
  },
  "edgemode": Object {
    "propertyName": "edgeMode",
  },
  "elevation": Object {
    "propertyName": "elevation",
  },
  "enable-background": Object {
    "propertyName": "enableBackground",
  },
  "enablebackground": Object {
    "propertyName": "enableBackground",
  },
  "end": Object {
    "propertyName": "end",
  },
  "exponent": Object {
    "propertyName": "exponent",
  },
  "externalResourcesRequired": Object {
    "propertyName": "externalResourcesRequired",
  },
  "externalresourcesrequired": Object {
    "propertyName": "externalResourcesRequired",
  },
  "fill": Object {
    "propertyName": "fill",
  },
  "fill-opacity": Object {
    "propertyName": "fillOpacity",
  },
  "fill-rule": Object {
    "propertyName": "fillRule",
  },
  "fillopacity": Object {
    "propertyName": "fillOpacity",
  },
  "fillrule": Object {
    "propertyName": "fillRule",
  },
  "filter": Object {
    "propertyName": "filter",
  },
  "filterRes": Object {
    "propertyName": "filterRes",
  },
  "filterUnits": Object {
    "propertyName": "filterUnits",
  },
  "filterres": Object {
    "propertyName": "filterRes",
  },
  "filterunits": Object {
    "propertyName": "filterUnits",
  },
  "flood-color": Object {
    "propertyName": "floodColor",
  },
  "flood-opacity": Object {
    "propertyName": "floodOpacity",
  },
  "floodcolor": Object {
    "propertyName": "floodColor",
  },
  "floodopacity": Object {
    "propertyName": "floodOpacity",
  },
  "focusable": Object {
    "propertyName": "focusable",
  },
  "font-family": Object {
    "propertyName": "fontFamily",
  },
  "font-size": Object {
    "propertyName": "fontSize",
  },
  "font-size-adjust": Object {
    "propertyName": "fontSizeAdjust",
  },
  "font-stretch": Object {
    "propertyName": "fontStretch",
  },
  "font-style": Object {
    "propertyName": "fontStyle",
  },
  "font-variant": Object {
    "propertyName": "fontVariant",
  },
  "font-weight": Object {
    "propertyName": "fontWeight",
  },
  "fontfamily": Object {
    "propertyName": "fontFamily",
  },
  "fontsize": Object {
    "propertyName": "fontSize",
  },
  "fontsizeadjust": Object {
    "propertyName": "fontSizeAdjust",
  },
  "fontstretch": Object {
    "propertyName": "fontStretch",
  },
  "fontstyle": Object {
    "propertyName": "fontStyle",
  },
  "fontvariant": Object {
    "propertyName": "fontVariant",
  },
  "fontweight": Object {
    "propertyName": "fontWeight",
  },
  "format": Object {
    "propertyName": "format",
  },
  "from": Object {
    "propertyName": "from",
  },
  "fx": Object {
    "propertyName": "fx",
  },
  "fy": Object {
    "propertyName": "fy",
  },
  "g1": Object {
    "propertyName": "g1",
  },
  "g2": Object {
    "propertyName": "g2",
  },
  "glyph-name": Object {
    "propertyName": "glyphName",
  },
  "glyph-orientation-horizontal": Object {
    "propertyName": "glyphOrientationHorizontal",
  },
  "glyph-orientation-vertical": Object {
    "propertyName": "glyphOrientationVertical",
  },
  "glyphRef": Object {
    "propertyName": "glyphRef",
  },
  "glyphname": Object {
    "propertyName": "glyphName",
  },
  "glyphorientationhorizontal": Object {
    "propertyName": "glyphOrientationHorizontal",
  },
  "glyphorientationvertical": Object {
    "propertyName": "glyphOrientationVertical",
  },
  "glyphref": Object {
    "propertyName": "glyphRef",
  },
  "gradientTransform": Object {
    "propertyName": "gradientTransform",
  },
  "gradientUnits": Object {
    "propertyName": "gradientUnits",
  },
  "gradienttransform": Object {
    "propertyName": "gradientTransform",
  },
  "gradientunits": Object {
    "propertyName": "gradientUnits",
  },
  "hanging": Object {
    "propertyName": "hanging",
  },
  "horiz-adv-x": Object {
    "propertyName": "horizAdvX",
  },
  "horiz-origin-x": Object {
    "propertyName": "horizOriginX",
  },
  "horizadvx": Object {
    "propertyName": "horizAdvX",
  },
  "horizoriginx": Object {
    "propertyName": "horizOriginX",
  },
  "ideographic": Object {
    "propertyName": "ideographic",
  },
  "image-rendering": Object {
    "propertyName": "imageRendering",
  },
  "imagerendering": Object {
    "propertyName": "imageRendering",
  },
  "in": Object {
    "propertyName": "in",
  },
  "in2": Object {
    "propertyName": "in2",
  },
  "intercept": Object {
    "propertyName": "intercept",
  },
  "k": Object {
    "propertyName": "k",
  },
  "k1": Object {
    "propertyName": "k1",
  },
  "k2": Object {
    "propertyName": "k2",
  },
  "k3": Object {
    "propertyName": "k3",
  },
  "k4": Object {
    "propertyName": "k4",
  },
  "kernelMatrix": Object {
    "propertyName": "kernelMatrix",
  },
  "kernelUnitLength": Object {
    "propertyName": "kernelUnitLength",
  },
  "kernelmatrix": Object {
    "propertyName": "kernelMatrix",
  },
  "kernelunitlength": Object {
    "propertyName": "kernelUnitLength",
  },
  "kerning": Object {
    "propertyName": "kerning",
  },
  "keyPoints": Object {
    "propertyName": "keyPoints",
  },
  "keySplines": Object {
    "propertyName": "keySplines",
  },
  "keyTimes": Object {
    "propertyName": "keyTimes",
  },
  "keypoints": Object {
    "propertyName": "keyPoints",
  },
  "keysplines": Object {
    "propertyName": "keySplines",
  },
  "keytimes": Object {
    "propertyName": "keyTimes",
  },
  "lengthAdjust": Object {
    "propertyName": "lengthAdjust",
  },
  "lengthadjust": Object {
    "propertyName": "lengthAdjust",
  },
  "letter-spacing": Object {
    "propertyName": "letterSpacing",
  },
  "letterspacing": Object {
    "propertyName": "letterSpacing",
  },
  "lighting-color": Object {
    "propertyName": "lightingColor",
  },
  "lightingcolor": Object {
    "propertyName": "lightingColor",
  },
  "limitingConeAngle": Object {
    "propertyName": "limitingConeAngle",
  },
  "limitingconeangle": Object {
    "propertyName": "limitingConeAngle",
  },
  "local": Object {
    "propertyName": "local",
  },
  "marker-end": Object {
    "propertyName": "markerEnd",
  },
  "marker-mid": Object {
    "propertyName": "markerMid",
  },
  "marker-start": Object {
    "propertyName": "markerStart",
  },
  "markerHeight": Object {
    "propertyName": "markerHeight",
  },
  "markerUnits": Object {
    "propertyName": "markerUnits",
  },
  "markerWidth": Object {
    "propertyName": "markerWidth",
  },
  "markerend": Object {
    "propertyName": "markerEnd",
  },
  "markerheight": Object {
    "propertyName": "markerHeight",
  },
  "markermid": Object {
    "propertyName": "markerMid",
  },
  "markerstart": Object {
    "propertyName": "markerStart",
  },
  "markerunits": Object {
    "propertyName": "markerUnits",
  },
  "markerwidth": Object {
    "propertyName": "markerWidth",
  },
  "mask": Object {
    "propertyName": "mask",
  },
  "maskContentUnits": Object {
    "propertyName": "maskContentUnits",
  },
  "maskUnits": Object {
    "propertyName": "maskUnits",
  },
  "maskcontentunits": Object {
    "propertyName": "maskContentUnits",
  },
  "maskunits": Object {
    "propertyName": "maskUnits",
  },
  "mathematical": Object {
    "propertyName": "mathematical",
  },
  "mode": Object {
    "propertyName": "mode",
  },
  "numOctaves": Object {
    "propertyName": "numOctaves",
  },
  "numoctaves": Object {
    "propertyName": "numOctaves",
  },
  "offset": Object {
    "propertyName": "offset",
  },
  "opacity": Object {
    "propertyName": "opacity",
  },
  "operator": Object {
    "propertyName": "operator",
  },
  "order": Object {
    "propertyName": "order",
  },
  "orient": Object {
    "propertyName": "orient",
  },
  "orientation": Object {
    "propertyName": "orientation",
  },
  "origin": Object {
    "propertyName": "origin",
  },
  "overflow": Object {
    "propertyName": "overflow",
  },
  "overline-position": Object {
    "propertyName": "overlinePosition",
  },
  "overline-thickness": Object {
    "propertyName": "overlineThickness",
  },
  "overlineposition": Object {
    "propertyName": "overlinePosition",
  },
  "overlinethickness": Object {
    "propertyName": "overlineThickness",
  },
  "paint-order": Object {
    "propertyName": "paintOrder",
  },
  "paintorder": Object {
    "propertyName": "paintOrder",
  },
  "panose-1": Object {
    "propertyName": "panose1",
  },
  "panose1": Object {
    "propertyName": "panose1",
  },
  "pathLength": Object {
    "propertyName": "pathLength",
  },
  "pathlength": Object {
    "propertyName": "pathLength",
  },
  "patternContentUnits": Object {
    "propertyName": "patternContentUnits",
  },
  "patternTransform": Object {
    "propertyName": "patternTransform",
  },
  "patternUnits": Object {
    "propertyName": "patternUnits",
  },
  "patterncontentunits": Object {
    "propertyName": "patternContentUnits",
  },
  "patterntransform": Object {
    "propertyName": "patternTransform",
  },
  "patternunits": Object {
    "propertyName": "patternUnits",
  },
  "pointer-events": Object {
    "propertyName": "pointerEvents",
  },
  "pointerevents": Object {
    "propertyName": "pointerEvents",
  },
  "points": Object {
    "propertyName": "points",
  },
  "pointsAtX": Object {
    "propertyName": "pointsAtX",
  },
  "pointsAtY": Object {
    "propertyName": "pointsAtY",
  },
  "pointsAtZ": Object {
    "propertyName": "pointsAtZ",
  },
  "pointsatx": Object {
    "propertyName": "pointsAtX",
  },
  "pointsaty": Object {
    "propertyName": "pointsAtY",
  },
  "pointsatz": Object {
    "propertyName": "pointsAtZ",
  },
  "preserveAlpha": Object {
    "propertyName": "preserveAlpha",
  },
  "preserveAspectRatio": Object {
    "propertyName": "preserveAspectRatio",
  },
  "preservealpha": Object {
    "propertyName": "preserveAlpha",
  },
  "preserveaspectratio": Object {
    "propertyName": "preserveAspectRatio",
  },
  "primitiveUnits": Object {
    "propertyName": "primitiveUnits",
  },
  "primitiveunits": Object {
    "propertyName": "primitiveUnits",
  },
  "r": Object {
    "propertyName": "r",
  },
  "radius": Object {
    "propertyName": "radius",
  },
  "refX": Object {
    "propertyName": "refX",
  },
  "refY": Object {
    "propertyName": "refY",
  },
  "refx": Object {
    "propertyName": "refX",
  },
  "refy": Object {
    "propertyName": "refY",
  },
  "rendering-intent": Object {
    "propertyName": "renderingIntent",
  },
  "renderingintent": Object {
    "propertyName": "renderingIntent",
  },
  "repeatCount": Object {
    "propertyName": "repeatCount",
  },
  "repeatDur": Object {
    "propertyName": "repeatDur",
  },
  "repeatcount": Object {
    "propertyName": "repeatCount",
  },
  "repeatdur": Object {
    "propertyName": "repeatDur",
  },
  "requiredExtensions": Object {
    "propertyName": "requiredExtensions",
  },
  "requiredFeatures": Object {
    "propertyName": "requiredFeatures",
  },
  "requiredextensions": Object {
    "propertyName": "requiredExtensions",
  },
  "requiredfeatures": Object {
    "propertyName": "requiredFeatures",
  },
  "restart": Object {
    "propertyName": "restart",
  },
  "result": Object {
    "propertyName": "result",
  },
  "rotate": Object {
    "propertyName": "rotate",
  },
  "rx": Object {
    "propertyName": "rx",
  },
  "ry": Object {
    "propertyName": "ry",
  },
  "scale": Object {
    "propertyName": "scale",
  },
  "seed": Object {
    "propertyName": "seed",
  },
  "shape-rendering": Object {
    "propertyName": "shapeRendering",
  },
  "shaperendering": Object {
    "propertyName": "shapeRendering",
  },
  "slope": Object {
    "propertyName": "slope",
  },
  "spacing": Object {
    "propertyName": "spacing",
  },
  "specularConstant": Object {
    "propertyName": "specularConstant",
  },
  "specularExponent": Object {
    "propertyName": "specularExponent",
  },
  "specularconstant": Object {
    "propertyName": "specularConstant",
  },
  "specularexponent": Object {
    "propertyName": "specularExponent",
  },
  "speed": Object {
    "propertyName": "speed",
  },
  "spreadMethod": Object {
    "propertyName": "spreadMethod",
  },
  "spreadmethod": Object {
    "propertyName": "spreadMethod",
  },
  "startOffset": Object {
    "propertyName": "startOffset",
  },
  "startoffset": Object {
    "propertyName": "startOffset",
  },
  "stdDeviation": Object {
    "propertyName": "stdDeviation",
  },
  "stddeviation": Object {
    "propertyName": "stdDeviation",
  },
  "stemh": Object {
    "propertyName": "stemh",
  },
  "stemv": Object {
    "propertyName": "stemv",
  },
  "stitchTiles": Object {
    "propertyName": "stitchTiles",
  },
  "stitchtiles": Object {
    "propertyName": "stitchTiles",
  },
  "stop-color": Object {
    "propertyName": "stopColor",
  },
  "stop-opacity": Object {
    "propertyName": "stopOpacity",
  },
  "stopcolor": Object {
    "propertyName": "stopColor",
  },
  "stopopacity": Object {
    "propertyName": "stopOpacity",
  },
  "strikethrough-position": Object {
    "propertyName": "strikethroughPosition",
  },
  "strikethrough-thickness": Object {
    "propertyName": "strikethroughThickness",
  },
  "strikethroughposition": Object {
    "propertyName": "strikethroughPosition",
  },
  "strikethroughthickness": Object {
    "propertyName": "strikethroughThickness",
  },
  "string": Object {
    "propertyName": "string",
  },
  "stroke": Object {
    "propertyName": "stroke",
  },
  "stroke-dasharray": Object {
    "propertyName": "strokeDasharray",
  },
  "stroke-dashoffset": Object {
    "propertyName": "strokeDashoffset",
  },
  "stroke-linecap": Object {
    "propertyName": "strokeLinecap",
  },
  "stroke-linejoin": Object {
    "propertyName": "strokeLinejoin",
  },
  "stroke-miterlimit": Object {
    "propertyName": "strokeMiterlimit",
  },
  "stroke-opacity": Object {
    "propertyName": "strokeOpacity",
  },
  "stroke-width": Object {
    "propertyName": "strokeWidth",
  },
  "strokedasharray": Object {
    "propertyName": "strokeDasharray",
  },
  "strokedashoffset": Object {
    "propertyName": "strokeDashoffset",
  },
  "strokelinecap": Object {
    "propertyName": "strokeLinecap",
  },
  "strokelinejoin": Object {
    "propertyName": "strokeLinejoin",
  },
  "strokemiterlimit": Object {
    "propertyName": "strokeMiterlimit",
  },
  "strokeopacity": Object {
    "propertyName": "strokeOpacity",
  },
  "strokewidth": Object {
    "propertyName": "strokeWidth",
  },
  "surfaceScale": Object {
    "propertyName": "surfaceScale",
  },
  "surfacescale": Object {
    "propertyName": "surfaceScale",
  },
  "systemLanguage": Object {
    "propertyName": "systemLanguage",
  },
  "systemlanguage": Object {
    "propertyName": "systemLanguage",
  },
  "tableValues": Object {
    "propertyName": "tableValues",
  },
  "tablevalues": Object {
    "propertyName": "tableValues",
  },
  "targetX": Object {
    "propertyName": "targetX",
  },
  "targetY": Object {
    "propertyName": "targetY",
  },
  "targetx": Object {
    "propertyName": "targetX",
  },
  "targety": Object {
    "propertyName": "targetY",
  },
  "text-anchor": Object {
    "propertyName": "textAnchor",
  },
  "text-decoration": Object {
    "propertyName": "textDecoration",
  },
  "text-rendering": Object {
    "propertyName": "textRendering",
  },
  "textLength": Object {
    "propertyName": "textLength",
  },
  "textanchor": Object {
    "propertyName": "textAnchor",
  },
  "textdecoration": Object {
    "propertyName": "textDecoration",
  },
  "textlength": Object {
    "propertyName": "textLength",
  },
  "textrendering": Object {
    "propertyName": "textRendering",
  },
  "to": Object {
    "propertyName": "to",
  },
  "transform": Object {
    "propertyName": "transform",
  },
  "u1": Object {
    "propertyName": "u1",
  },
  "u2": Object {
    "propertyName": "u2",
  },
  "underline-position": Object {
    "propertyName": "underlinePosition",
  },
  "underline-thickness": Object {
    "propertyName": "underlineThickness",
  },
  "underlineposition": Object {
    "propertyName": "underlinePosition",
  },
  "underlinethickness": Object {
    "propertyName": "underlineThickness",
  },
  "unicode": Object {
    "propertyName": "unicode",
  },
  "unicode-bidi": Object {
    "propertyName": "unicodeBidi",
  },
  "unicode-range": Object {
    "propertyName": "unicodeRange",
  },
  "unicodebidi": Object {
    "propertyName": "unicodeBidi",
  },
  "unicoderange": Object {
    "propertyName": "unicodeRange",
  },
  "units-per-em": Object {
    "propertyName": "unitsPerEm",
  },
  "unitsperem": Object {
    "propertyName": "unitsPerEm",
  },
  "v-alphabetic": Object {
    "propertyName": "vAlphabetic",
  },
  "v-hanging": Object {
    "propertyName": "vHanging",
  },
  "v-ideographic": Object {
    "propertyName": "vIdeographic",
  },
  "v-mathematical": Object {
    "propertyName": "vMathematical",
  },
  "valphabetic": Object {
    "propertyName": "vAlphabetic",
  },
  "values": Object {
    "propertyName": "values",
  },
  "vector-effect": Object {
    "propertyName": "vectorEffect",
  },
  "vectoreffect": Object {
    "propertyName": "vectorEffect",
  },
  "version": Object {
    "propertyName": "version",
  },
  "vert-adv-y": Object {
    "propertyName": "vertAdvY",
  },
  "vert-origin-x": Object {
    "propertyName": "vertOriginX",
  },
  "vert-origin-y": Object {
    "propertyName": "vertOriginY",
  },
  "vertadvy": Object {
    "propertyName": "vertAdvY",
  },
  "vertoriginx": Object {
    "propertyName": "vertOriginX",
  },
  "vertoriginy": Object {
    "propertyName": "vertOriginY",
  },
  "vhanging": Object {
    "propertyName": "vHanging",
  },
  "videographic": Object {
    "propertyName": "vIdeographic",
  },
  "viewBox": Object {
    "propertyName": "viewBox",
  },
  "viewTarget": Object {
    "propertyName": "viewTarget",
  },
  "viewbox": Object {
    "propertyName": "viewBox",
  },
  "viewtarget": Object {
    "propertyName": "viewTarget",
  },
  "visibility": Object {
    "propertyName": "visibility",
  },
  "vmathematical": Object {
    "propertyName": "vMathematical",
  },
  "widths": Object {
    "propertyName": "widths",
  },
  "word-spacing": Object {
    "propertyName": "wordSpacing",
  },
  "wordspacing": Object {
    "propertyName": "wordSpacing",
  },
  "writing-mode": Object {
    "propertyName": "writingMode",
  },
  "writingmode": Object {
    "propertyName": "writingMode",
  },
  "x": Object {
    "propertyName": "x",
  },
  "x-height": Object {
    "propertyName": "xHeight",
  },
  "x1": Object {
    "propertyName": "x1",
  },
  "x2": Object {
    "propertyName": "x2",
  },
  "xChannelSelector": Object {
    "propertyName": "xChannelSelector",
  },
  "xchannelselector": Object {
    "propertyName": "xChannelSelector",
  },
  "xheight": Object {
    "propertyName": "xHeight",
  },
  "xlink:actuate": Object {
    "propertyName": "xlinkActuate",
  },
  "xlink:arcrole": Object {
    "propertyName": "xlinkArcrole",
  },
  "xlink:href": Object {
    "propertyName": "xlinkHref",
  },
  "xlink:role": Object {
    "propertyName": "xlinkRole",
  },
  "xlink:show": Object {
    "propertyName": "xlinkShow",
  },
  "xlink:title": Object {
    "propertyName": "xlinkTitle",
  },
  "xlink:type": Object {
    "propertyName": "xlinkType",
  },
  "xlinkactuate": Object {
    "propertyName": "xlinkActuate",
  },
  "xlinkarcrole": Object {
    "propertyName": "xlinkArcrole",
  },
  "xlinkhref": Object {
    "propertyName": "xlinkHref",
  },
  "xlinkrole": Object {
    "propertyName": "xlinkRole",
  },
  "xlinkshow": Object {
    "propertyName": "xlinkShow",
  },
  "xlinktitle": Object {
    "propertyName": "xlinkTitle",
  },
  "xlinktype": Object {
    "propertyName": "xlinkType",
  },
  "xml:base": Object {
    "propertyName": "xmlBase",
  },
  "xml:lang": Object {
    "propertyName": "xmlLang",
  },
  "xml:space": Object {
    "propertyName": "xmlSpace",
  },
  "xmlbase": Object {
    "propertyName": "xmlBase",
  },
  "xmllang": Object {
    "propertyName": "xmlLang",
  },
  "xmlns": Object {
    "propertyName": "xmlns",
  },
  "xmlns:xlink": Object {
    "propertyName": "xmlnsXlink",
  },
  "xmlnsxlink": Object {
    "propertyName": "xmlnsXlink",
  },
  "xmlspace": Object {
    "propertyName": "xmlSpace",
  },
  "y": Object {
    "propertyName": "y",
  },
  "y1": Object {
    "propertyName": "y1",
  },
  "y2": Object {
    "propertyName": "y2",
  },
  "yChannelSelector": Object {
    "propertyName": "yChannelSelector",
  },
  "ychannelselector": Object {
    "propertyName": "yChannelSelector",
  },
  "z": Object {
    "propertyName": "z",
  },
  "zoomAndPan": Object {
    "propertyName": "zoomAndPan",
  },
  "zoomandpan": Object {
    "propertyName": "zoomAndPan",
  },
}
`;
